### PR TITLE
TST/CI: Add automated testing for pull requests

### PR DESCRIPTION
**Pull request by greglucas**
Add pyproject.toml so we can install the project
Add GitHub Actions CI
Add network test markers to ignore tests that require network connectivity

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+[build-system]
```
Comment (tech3371) [inline pyproject.toml:1]: I haven't used toml file before. Are these value referenced by something?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (tech3371) [inline .github/workflows/test.yml:7]: I see. Just stating pull_request allows it to run test when anybody creates PR. Didn't know that before.

**Review by tech3371 — APPROVED** [review]
Looks good to me!

File: pyproject.toml (hunk 1 above)
Comment (greglucas) [reply pyproject.toml:1]: This is the way to specify your project setup now and can reference other tools and requirements as well `[tool.xyz]`. For example, poetry would look different than this.
https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

https://peps.python.org/pep-0517/

**Review by bryan-harter — APPROVED** [review]

**Pull request merged**